src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValue.cs

```diff
@@ -38,13 +40,46 @@ public abstract partial class JsonValue : JsonNode
                     return null;
                 }
 
-                if (element.ValueKind == JsonValueKind.Object || element.ValueKind == JsonValueKind.Array)
+                VerifyJsonElementIsNotArrayOrObject(ref element);
+            }
+
+            return new JsonValueNotTrimmable<T>(value, options);
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="JsonValue"/> class that contains the specified value.
+        /// </summary>
+        /// <returns>
+        ///   The new instance of the <see cref="JsonValue"/> class that contains the specified value.
+        /// </returns>
+        /// <typeparam name="T">The type of value to create.</typeparam>
+        /// <param name="value">The value to create.</param>
+        /// <param name="jsonTypeInfo">The <see cref="JsonTypeInfo"/> that is later used to serialize the value.</param>
```

Comment:
It might be better to say 

`The <see cref="JsonTypeInfo"/> that will be used to serialize the value.`